modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk/query_processor_test.clj

```diff
@@ -373,23 +373,57 @@
             (is (= :completed
                    (:status (qp/process-query query))))))))))
 
+(deftest temporal-type-conversion-test
+  (mt/with-driver :bigquery-cloud-sdk
+    (mt/with-temporary-setting-values [report-timezone "US/Pacific"]
+      (let [temporal-string "2022-01-01"
+            convert         (fn [from-t to-t]
+                              (->> (#'bigquery.qp/->temporal-type to-t (#'bigquery.qp/->temporal-type from-t temporal-string))
+                                   (sql.qp/format-honeysql :bigquery-cloud-sdk)))]
+        (testing "convert from datetime to different temporal types"
+          (testing :time
+            (is (= ["time(datetime(?))" temporal-string]
+                   (convert :datetime :time))))
+          (testing :date
+            (is (= ["date(datetime(?))" temporal-string]
+                   (convert :datetime :date))))
+          (testing :timestamp
+            (is (= ["timestamp(datetime(?), 'US/Pacific')" temporal-string]
```

Comment:
Would it make sense to use the report timezone instead of the literal `US/Pacific`?

Reply:
Maybe? I'm not sure that would make it more readable. It would make it easier to change report-timezone to something else, but it's not necessary in this case. Do you have an explanation?

Reply:
The construction would be less readable but the meaning would be clearer, I think. As it stands now, the reader has to look back and search to see that this is in fact the reporting timezone. Also, if we change the reporting timezone for any reason, we would change it once. But it's not a big deal.